test/minikube/minikube_sm_test.go

```diff
@@ -0,0 +1,291 @@
+// +build long
+
+/*
+ * Tests to verify operation of SM startup.
+ * Separate so that changes to SM startup in general - and the SM startup scipt(s) in particular
+ * can be tested in isolation.
+ */
+
+package minikube
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/nuodb/nuodb-helm-charts/test/testlib"
+	"github.com/stretchr/testify/assert"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+func TestKubernetesStartSM(t *testing.T) {
+	// skip this test until the code to detect and recover from database startup failure is robust
+	t.Skip()
+
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+
+	adminOptions := helm.Options{}
+
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+	helmChartReleaseName, namespaceName := testlib.StartAdmin(t, &adminOptions, 1, "")
+
+	admin0 := fmt.Sprintf("%s-nuodb-cluster0-0", helmChartReleaseName)
+
+	t.Run("startDatabaseStatefulSet", func(t *testing.T) {
+		defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+		databaseOptions := helm.Options{
+			SetValues: map[string]string{
+				"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+				"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+				"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+				"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+				"backup.persistence.enabled":            "true",
+				"backup.persistence.size":               "1Gi",
+				"database.env[0].name":                  "NUODB_DEBUG",
+				"database.env[0].value":                 "debug",
+			},
+		}
+
+		databaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &databaseOptions)
+
+		fmt.Println("startDatabase returned...")
+
+		opt := testlib.GetExtractedOptions(&databaseOptions)
+		smPodTemplate := fmt.Sprintf("sm-%s-nuodb-%s-%s", databaseChartName, opt.ClusterName, opt.DbName)
+		sm0 := testlib.GetPodName(t, namespaceName, smPodTemplate)
+
+		fmt.Println("extracted databaseOptions; sm name is ", sm0)
+
+		kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+
+		testlib.AddDiagnosticTeardown(
+			testlib.TEARDOWN_DATABASE,
+			t,
+			func() { testlib.GetFile(t, namespaceName, sm0, "/var/log/nuodb", "nuosm.log") },
+		)
+
+		defer func() {
+			status, _ := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", admin0, "--", "nuocmd", "show", "database", "--db-name", opt.DbName)
+			fmt.Println("database status", status)
```

Comment:
t.log. But you dont really need it. RunKubectlAndGetOutput prints everything to the log anyways. You will be printing everything twice.

Reply:
I was adding context, so I could trace the evolution of the database state.